mantidproject/mantid
# NormaliseToMonitor Detector Scan and Single Count Workspaces

These changes have 2 significant parts
 * Supporting workspaces with scanning detectors  - here there are more than one monitor spectrum to normalise to, so the input must accept the MonitorID only
 * Supporting workspaces with point data and single counts

**To test:**

Check various inputs, point data with more than one bin should fail, all else should pass.

Try loading the attached file for D2B, and check normalisation works correctly (monitor values and detector are artificially changed in this file to help test [508093_monitors_changed.nxs.zip](https://github.com/mantidproject/mantid/files/1375143/508093_monitors_changed.nxs.zip). This file is both scanning and single count per spectrum.

Further tests could be done with `CreateSampleWorkspace`, using the option to set the number of scan points and/or making a single bin workspace, then converting it to point data. Tests can also be performed for the previously supported cases, but these should have been picked up by the unit tests.

Fixes #20619.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
